cargo-miri/bin.rs

```diff
@@ -510,8 +530,59 @@ fn phase_cargo_miri(mut args: env::Args) {
         &host
     };
 
-    // Forward all further arguments to cargo.
-    cmd.args(args);
+    let mut target_dir = None;
+
+    // Forward all arguments before `--` other than `--target-dir` and its value to Cargo.
+    for arg in ArgFlagValueWithOtherArgsIter::new(&mut args, "--target-dir") {
+        match arg {
+            Ok(value) => target_dir = Some(value.into()),
+            Err(arg) => drop(cmd.arg(arg)),
+        }
+    }
+
+    // Detect the target directory if it's not specified via `--target-dir`.
+    let target_dir = target_dir.get_or_insert_with(|| {
+        #[derive(Deserialize)]
+        struct Metadata {
+            target_directory: PathBuf,
+        }
+        let mut cmd = cargo();
+        // `-Zunstable-options` is required by `--config`.
+        cmd.args(["metadata", "--no-deps", "--format-version=1", "-Zunstable-options"]);
+        // The `build.target-dir` config can by passed by `--config` flags, so forward them to
```

Comment:
```suggestion
        // The `build.target-dir` config can be bypassed by `--config` flags, so forward them to
```

Reply:
Is "bypassed" the right word...? https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#config-cli describes `--config` as "allowing arbitrary config values to be passed in via the command-line", so my understanding is that it can bypass `build.target-dir` in config files (because Cargo merges values of the same config key in different places), but it does not bypass the config itself.

Reply:
You wrote "by passed" so I thought you meant "bypassed" but maybe you just meant "be passed"